tools/eslint-rules/snake-case-filenames.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @fileoverview enforces snake_case for filenames.
+ * @author Shobhit Chittora <chittorashobhit@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const path = require('path');
+
+module.exports = function(context) {
+  const snakeCaseRegex = new RegExp('^[a-z_]+$');
+  const targetDirs = ['lib'];
```

Comment:
Do not hardcode the directory.

Reply:
To elaborate on this: If I enable this rule in a config file in the `tools` directory, hardcoding the directory names in the code here means that the rule won't check anything in the `tools` directory. This is not the behavior we'd want.